S/SuiteSparse@5/build_tarballs.jl

```diff
@@ -52,43 +47,47 @@ done
 
 # For now, we'll have to adjust the name of the OpenBLAS library on macOS.
 # Eventually, this should be fixed upstream
-if [[ ${target} == "x86_64-apple-darwin14" ]]; then
+if [[ ${target} == "*-apple-*" ]]; then
     echo "-- Modifying library name for OpenBLAS"
 
     for nm in libcholmod.3.0.13 libspqr.2.0.9 libumfpack.5.7.8; do
-        install_name_tool -change libopenblas64_.0.3.3.dylib @rpath/libopenblas64_.dylib ${prefix}/lib/${nm}.dylib
+        # Figure out what version it probably latched on to:
+        OPENBLAS_LINK=$(otool -L ${libdir}/${nm}.dylib | grep libopenblas64_ | awk '{ print $1 }')
+        install_name_tool -change ${OPENBLAS_LINK} @rpath/libopenblas64_.dylib ${libdir}/${nm}.dylib
     done
 fi
 
 # Compile suitesparse_wrapper shim
 cd $WORKSPACE/srcdir/SuiteSparse_wrapper
 make "${FLAGS[@]}" install
+
+install_license ${WORKSPACE}/srcdir/SuiteSparse/LICENSE.txt
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_gcc_versions(supported_platforms())
+platforms = expand_cxxstring_abis(supported_platforms())
```

Comment:
I'm pretty sure SuiteSparse has fortran code within it, so we need to expand the libgfortran versions.  I don't think SuiteSparse has a C++ ABI, so we shouldn't need to expand the cxx string ABIs.

Reply:
I was confused also because https://github.com/DrTimothyAldenDavis/SuiteSparse shows C and C++ as main languages.  I see that there are Fortran files, they must account for less than 1% of the total code, but I guess they are the important bit for us here?

Reply:
I'm pretty sure that we don't use any C++ interfaces, so it's _probably_ fine to ignore the C++ stuff.  And yes, the libgfortran stuff is way more finnicky than the C++ stuff.

Reply:
There's no Fortran in SuiteSparse. It calls the BLAS, and perhaps LAPACK. It looks like we are using the C-callable APIs only.

Reply:
`libsuitesparse_wrapper.so`, `libumfpack.so`, `libcholmod.so` and `libspqr.so` all link against `libgfortran`. If it's not because of fortran code, it's because somebody is explicitly passing `-lgfortran` to the linker.  I don't know why they would do that, but it means that we have to expand libgfortran versions.

Reply:
Let me check if that is required.

Reply:
Do you need to pass `-lgfortran` in order to link against openblas?

Reply:
I don't see an explicit `-lgfortran` in the build file. Any ideas where it is coming from?

Reply:
It's in the SuiteSparse source.  I don't know why.  Perhaps because the author doesn't know to use `-Wl,-rpath-link,$(libgfortran_dir)` to get `gcc` to find `libgfortran` when only passed `-lopenblas`?

```
$ grep lgfortran -r .
./UMFPACK/Tcov/Make.7:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.8:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.1:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.6:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.3:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.4:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.5:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./UMFPACK/Tcov/Make.2:LDLIBS = -L$(INSTALL)/lib -llapack -lrefblas -lgfortran -lm -lrt \
./SPQR/Demo/Makefile:# FLIB = -lgfortran -llapack_plain -lblas_plain -lg2c
./SPQR/Tcov/Makefile:# FLIB = -llapack_plain -lblas_plain -lgfortran
./SPQR/Tcov/Makefile:# FLIB = -llapack_plain -lblas_plain -lgfortran -lg2c
./Mongoose/SuiteSparse_config/SuiteSparse_config.mk:        BLAS = -lrefblas -lgfortran -lstdc++
./Mongoose/SuiteSparse_config/SuiteSparse_config.mk:    #   make BLAS='-lacml -lgfortran'
./Mongoose/SuiteSparse_config/SuiteSparse_config.mk:        ## BLAS = -lrefblas -lgfortran
./SuiteSparse_config/SuiteSparse_config.mk:        BLAS = -lrefblas -lgfortran -lstdc++
./SuiteSparse_config/SuiteSparse_config.mk:    #   make BLAS='-lacml -lgfortran'
./SuiteSparse_config/SuiteSparse_config.mk:        ## BLAS = -lrefblas -lgfortran
./CHOLMOD/Tcov/Makefile:BLAS = -lrefblas -lgfortran
./CHOLMOD/Tcov/Makefile:# LDLIBS = $(METIS) -lm -llapack_plain -lblas_plain -lgfortran -lgfortranbegin
./CHOLMOD/Tcov/Makefile:# LDLIBS = $(METIS) -lm -llapack -lblas -lgfortran -lgfortranbegin
./CHOLMOD/Tcov/Makefile:# LDLIBS = $(METIS) -lm -llapack_plain -lblas_plain -lgfortran -lgfortranbegin -lpthread -lrt
./CHOLMOD/Tcov/Makefile:# LDLIBS = $(METIS) -lm -llapack -lblas -lgfortran -lgfortranbegin -lpthread
```

Reply:
That's all refblas stuff. I am pretty sure we override it all in our suitesparse build.

Reply:
I also see no `-lgfortran` in the azure pipelines build logs for suitesparse.

Reply:
Good news; you're right!  It looks like it all gets overridden, and the links that `ldd` were finding are all transitive through `libopenblas`.  This means that we should be able to not expand at all!  Huzzah for smaller build footprints!  Let's try it and see if the BB auditor complains at all.

Reply:
In fact we should remove all that CFOPENMP stuff if we can, and even try to set F77 to empty so that it doesn't accidentally find a fortran compiler.

Reply:
`CFOPENMP` is cleared for all platforms:
https://github.com/giordano/Yggdrasil/blob/35b51829bf6093869f6b0741ff88f10e412b5dcc/S/SuiteSparse%405/build_tarballs.jl#L21